### PR TITLE
bump softnpu

### PR DESCRIPTION
**Pull request by rcgoodfellow**
holding in draft until softnpu changes land

### CODE REVIEW

**Comment by pfmooney** [comment]
It looks like I should update the `clippy` xtask to specify `--no-deps`?

**Comment by rcgoodfellow** [comment]
> It looks like I should update the Clippy task to specify --no-deps?

Perhaps. This is pulling in `libnet`, which will go badly on any platform which is not illumos. Looks like some of the clippy complaints would also apply to illumos, so I'll go check that out.

File: .github/buildomat/jobs/falcon-build.sh [hunk 1]
```diff
@@ -18,6 +18,7 @@ rustc --version
 
 banner check
 ptime -m cargo check --features falcon
+ptime -m cargo xtask clippy
```
Comment (pfmooney) [inline .github/buildomat/jobs/falcon-build.sh:21]: Probably just scope this to a `cargo clippy ---features falcon`?  We don't need to run the non-falcon checks.

File: xtask/src/task_clippy.rs [hunk 2]
```diff
@@ -39,14 +39,17 @@ pub(crate) fn cmd_clippy(strict: bool, quiet: bool) -> Result<()> {
...
+    {
+        failed |= run_clippy(&[
+            "--features",
+            "falcon",
+            "-p",
+            "propolis-server",
+            "-p",
+            "propolis-client",
+        ])?;
+    }
```
Comment (pfmooney) [inline xtask/src/task_clippy.rs:52]: I believe this part of the change can be elided now that oxidecomputer/netadm-sys@d44d9e084f39e844f8083d4d9b39a331061ebbcc has landed?  (Provided a `cargo update -p libnet` is done)

Comment (rcgoodfellow) [reply xtask/src/task_clippy.rs]: Removed

File: .github/buildomat/jobs/falcon-build.sh (hunk 1 above)
Comment (rcgoodfellow) [reply .github/buildomat/jobs/falcon-build.sh]: done, also with `--all-targets` flag

**Review by pfmooney — APPROVED** [review]

**Pull request merged**